src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -112,28 +168,43 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.filter(!_.omitGraphML).foreach { c => c.edgesGraphML(buf, pad) }
   }
 
+  /** execute function to [[children]].
+    * @param iterfunc function to run
+    * */
   def nodeIterator(iterfunc: (LazyModule) => Unit): Unit = {
     iterfunc(this)
     children.foreach( _.nodeIterator(iterfunc) )
   }
 
+  /** get [[children]] */
   def getChildren = children
 }
 
 object LazyModule
 {
+  /** current [[LazyModule]] scope, specifically, it is a stack of [[LazyModule]].
+    * it will be dynamically set by [[LazyScope.apply]] and [[LazyModule.apply]],
+    * */
   protected[diplomacy] var scope: Option[LazyModule] = None
+  /** index of [[LazyModule]], notice there is no No.0 module. */
   private var index = 0
 
+  /** add a [[LazyModule]]
+    * @param bc [[LazyModule]] needed to be instantiated.
+    * @param valName [[ValName]] used to naming this val.
```

Comment:
```suggestion
    * @param valName [[ValName]] used to naming this module.
```

... or is it `naming this instance`?

Reply:
If not specifically assigned, it will be generated by `ValName` by default,

Reply:
I changed to 
```
    * @param valName [[ValName]] used to naming this val,
    *                it can be automatically generated by [[ValName]] macro, or specify manually.
```
Is better now?